plugins/org.eclipse.n4js/src/org/eclipse/n4js/resource/PostProcessingAwareResource.java

```diff
@@ -143,6 +143,30 @@ public void performPostProcessing() {
 	 * post-processing has already been performed <u>or is currently in progress</u>, so unnecessarily calling this
 	 * method should not do any major harm.
 	 * <p>
+	 * The precise effect of this method can be summarized as follows:
+	 * <table border="1">
+	 * <tr>
+	 * <th>State <em>before</em> this method is invoked</th>
+	 * <th>State <em>after</em> this method returns</th>
+	 * </tr>
+	 * <tr>
+	 * <td>post-processing not started yet</td>
+	 * <td>post-processing completed (!!)<br>
+	 * (call to {@link #performPostProcessing()} started post-processing on 'res' and completed it before
+	 * returning)</td>
+	 * </tr>
+	 * <tr>
+	 * <td>post-processing in progress</td>
+	 * <td>post-processing in progress<br>
+	 * (call to #performPostProcessing() was ignored, i.e. had no effect)</td>
```

Comment:
`{@link #performPostProcessing()}` for two remaining cases as well?

Reply:
done